### ISSUE TITLE
Support for creation of premium SKU eventhubs in Azure regions without Availability Zones

### ISSUE THREAD

**Issue by wSchweinsberg** [hashicorp/terraform-provider-azurerm#17291]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently premium SKU eventhubs can be created via Terraform only in regions where availability zones are enabled. 
Our team is trying to create premium SKU eventhubs in "West US" and cannot because Terraform requires premium SKU eventhubs to have `zone_redundancy = true`. Since the "West US" region doesn't have availability zones enabled, this blocks the creation of premium SKU eventhubs, 

Is it possible to have Terraform allow the creation of premium SKU eventhubs without setting `zone_redundant = true`?

### New or Affected Resource(s)/Data Source(s)

azurerm_eventhub_namespace

### Potential Terraform Configuration

```hcl
resource "azurerm_eventhub_namespace" "name" {
  count                         = var.resource-group-location
  name                          = "premium-eh"
  location                      = var.resource-group-location
  resource_group_name           = var.resource-group-name
  sku                           = "Premium"
  zone_redundant                = "false" # Currently needs to be set to true for creation of Premium SKU
  capacity                      = 2
  auto_inflate_enabled          = false
  tags                          = var.resource-group-tags
}
```

### References

https://docs.microsoft.com/en-us/azure/availability-zones/az-overview

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/eventhub_namespace

**Comment by xiaxyi** [comment]
@wSchweinsberg Let me check if there is any restriction from the api side, I'll update once confirmed

**Comment by xiaxyi** [comment]
Thanks guys for your patience on this issue, you may track the status in the mentioned pr.

**Comment by xiaxyi** [comment]
The zone_redundant will be enabled by default if we are choosing a zone that supports availability zone. Let me figure out how to better implement this.

**Comment by 2hat-roberio** [comment]
Any updates ?

**Comment by rgtoffolo** [comment]
+1 ... any updates ?

**Comment by djsly** [comment]
@xiaxyi can we move with your initial implementation for now, for the next release ?
I'm also looking for a fix for this: https://github.com/hashicorp/terraform-provider-azurerm/issues/17897 

I could help if the PR are simple

**Comment by xiaxyi** [comment]
Thanks @djsly, @wSchweinsberg and guys, you can track the status in the mentioned pr.

**Comment by djsly** [comment]
thanks @xiaxyi !

**Comment by djsly** [comment]
this is in v3.25.0

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #18542. As such, I am going to mark this issue as closed.
